packages/fuels/src/cli/config/loadConfig.ts

```diff
@@ -44,6 +44,10 @@ export async function loadConfig(cwd: string): Promise<FuelsConfig> {
   const useBuiltinForc = userConfig.useBuiltinForc ?? shouldUseBuiltinForc();
   const useBuiltinFuelCore = userConfig.useBuiltinFuelCore ?? shouldUseBuiltinFuelCore();
 
+  const { forcBuildFlags = [] } = userConfig;
+  const releaseFlag = forcBuildFlags.find((f) => f === '--release');
+  const buildMode = releaseFlag ? 'release' : 'debug';
```

Comment:
From what I understood, `buildMode` is only being used in `forcUtils` to get the correct path. If you were to combine the utils that are using the path into one function that returns an object, similar to what `getForcProject` does, you could move this `buildMode` extraction logic out from `loadConfig` and into that function and remove the `buildMode` property from `FuelsConfig`:

```ts
export function getPaths(contractPath: string, { forcBuildFlags: [] }: FuelsConfig){
  const releaseFlag = forcBuildFlags.find((f) => f === '--release');
  const buildMode = releaseFlag ? 'release' : 'debug';
  const outDir = join(contractPath, "out", buildMode);
  
  const projectName = getContractName(contractPath);

  return {
    binaryPath: join(outDir,`${projectName}.bin`),
    storageSlotsPath: join(outDir, `${projectName}-storage_slots.json`),
    abiPath: join(outDir, `${projectName}-abi.json`),
        
  }
}
```

I think that's worth the effort because it removes maintainer questions/ambiguity around these scenarios:

```ts
const fuelsConfig: FuelsConfig = {
  // ...,
  forcBuildFlags: ["--release"],
  buildMode: 'debug',
}

const fuelsConfig: FuelsConfig = {
  // ...,
  forcBuildFlags: ["--debug"],
  buildMode: 'release',
}
```

Reply:
Thanks for the analysis.

There is a clear distinction between `UserFuelsConfig` and `FuelsConfig`; the latter is not under the user domain, and we conveniently initialize it only once, immediately after loading the user config. The pattern is borrowed from the `basePath` and `configPath` props that follow the same principle.

Handing all configuration options in a single place provides a clear separation of concerns between configuration and utilities, and having the type clearly defined in the `types.ts` makes it easier to follow, not harder.

The conflicting scenario you described with `debug` x `release` seems made up and purely theoretical since it can never occur in practical terms, only mentally as a hypothesis. We initialize things _only once_ and they are tied together. Full test coverage also guarantees no silly mistakes will slip through.

Reply:
> The conflicting scenario you described with debug x release seems made up and purely theoretical since it can never occur in practical terms, only mentally as a hypothesis. We initialize things only once and they are tied together. Full test coverage also guarantees no silly mistakes will slip through.

It's not made up, I questioned why it exists and then went to analyze it and produced the suggestion as a way to reduce the ambiguity so that the question doesn't pop up anymore.

Reply:
Made up in the sense this can never occur, only as a mental hypothesis.

```ts
  forcBuildFlags: ["--debug"],
  buildMode: 'release',
```

The same hypothesis would hold regardless of where we'd put his block of code:

```ts
  const releaseFlag = forcBuildFlags.find((f) => f === '--release');
  const buildMode = releaseFlag ? 'release' : 'debug';
```